### DISCUSSION TITLE
Problema para acessar o BD  ao executar a API do curso 2812 (Azure)

### DISCUSSION THREAD

**Discussion by alexandrefreitasp** [balta-io/forum#213] [Tira Dúvidas] [answered]
Fiz a publicação de acordo com o curso e tudo correu perfeitamente, inclusive foi criado o BD e tabelas na Azure com "Action". O problema ocorre quando tento consumir a API via Postman e independente do método utilizado sempre recebo erro 500     "errors": [
        "05X01 - Falha interna no servidor"
    ]
}
Testei a API localmente apontando para o BD na Azure e funcionou corretamente. Acredito que por aguma razão ao tentar executar a aplicação ela não tenha permissão para acessar o BD mas o estranho é que para a crição do mesmo, correu tudo bem.

Detalhe: Foi criada a regra de firewall "ghactions"

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @alexandrefreitasp, erro 500 é interno do servidor... ou seja, deu algum erro na sua API.
Qual endpoint você chamou para testar? Tem como testar outra rota ou deixar uma rota mais simples, sem acesso a banco por exemplo?

Em adicional, você pode usar `try` no código para pegar possíveis erros e previnir este tipo de comportamento.

**Comment by alexandrefreitasp** [comment]
"errors": [
        "05X01 - Falha interna no servidor - Cannot open server 'xxxxxxxx-sqlserver' requested by the login. Client with IP address '20.xxxx0.75' is not allowed to access the server.  To enable access, use the Windows Azure Management Portal or run sp_set_firewall_rule on the master database to create a firewall rule for this IP address or address range.  It may take up to five minutes for this change to take effect."
    ]

Boa tarde, como sugerido, adicionei a exception ao retorno da API e recebi esse retorno. Vi que o range de IP é diferente do adicionado via workflow que está como ghactions                         104.xx.xx4.x77.Como devo proceder? Obrigado!

**Reply by andrebaltieri** [reply]
Boa tarde, @alexandrefreitasp, como vai?
Você precisa liberar este IP no firewall do seu servidor SQL... o App e SQL estão no mesmo resource group?

**Comment by alexandrefreitasp** [comment]
Sim, estou no mesmo resource group. Adicionei a regra e funcionou. Mas gostaria de entender o pq da necessidade de se adicionar o IP após a publicação.

**Reply by andrebaltieri** [reply]
@alexandrefreitasp, como vai? Até onde sei (E fiz no curso) o mesmo resource group não precisa dessa liberação. Você seguiu certinho as aulas? Consegue dar uma revisada ver se não faltou algo...

Foi estranho ele pedir esta liberação.

**Comment by alexandrefreitasp** [comment]
Oi, tudo bem e vc? Sim, segui todos os passos e os refiz algumas vezes tb. Realmente foi estranho ele pedir essa liberação.  De qualquer maneira vou acompanhar as publicações de novos projetos. Obrigado!

**Comment by thiagocajadev** [comment]
Boa noite pessoal. Tive o mesmo problema no Azure.

Para resolver, marquei a opção Exceções -> **Permitir que serviços e recursos do Azure acessem este servidor**.

Caminho completo: Bancos de dados SQL -> Configurar acesso -> Rede -> Exceções.

**Reply by andrebaltieri** [reply]
Muito obrigado pelo feedback!
@alexandrefreitasp 👆
